modify model to enable loading by blob

ChunliF opened the pull request:
Summary:
This script is used to generate a model with bound shape inference and
blob reorder, which are requirements for big model loading on T17.
1. Load existing model.
2. Do bound shape inference and blob reorder (put embedding blobs at the end).
3. Save the modified model.

Test Plan:
Generated a new moel and tested on NNPI.
P124181047 (mismatch is AA variance)

Differential Revision: D19165467

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 7d3d80c:


* **1/1** broken upstream at merge base 348d421 (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=348d42114ea8b5c85288a0f6d80789f97089f29c&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
  <details>
  <summary>
  You may want to rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  
* **0/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=7d3d80c3d627c1ab0a7b5f31b5728bde76f9a728).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [pytorch_linux_xenial_py3_clang5_asan_test](https://dr.pytorch.org/build-details.html?build_id=77979234)



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

ChunliF committed:
modify model to enable loading by blob (#31507)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/31507

This script is used to generate a model with bound shape inference and
blob reorder, which are requirements for big model loading on T17.
1. Load existing model.
2. Do bound shape inference and blob reorder (put embedding blobs at the end).
3. Save the modified model.

Test Plan:
Generated a new moel and tested on NNPI.
P124181047 (mismatch is AA variance)

Reviewed By: ipiszy

Differential Revision: D19165467

fbshipit-source-id: ad0d5e264cb780f27aa14f350902948fe2172b30
